custom_components/waste_collection_schedule/waste_collection_schedule/source/ipswich_qld_gov_au.py

```diff
@@ -0,0 +1,116 @@
+import datetime
+import urllib
+from html.parser import HTMLParser
+import requests
+
+from waste_collection_schedule import Collection
+
+TITLE = "Ipswich City Council"
+DESCRIPTION = "Source for Ipswich City Council rubbish collection."
+URL = "https://www.ipswich.qld.gov.au/live/waste-and-recycling/bin-collection-calendar"
+TEST_CASES = {
+    "Camira State School": {"street": "184-202 Old Logan Rd", "suburb": "Camira"},
+    "Random": {"street":"50 Brisbane Road", "suburb": "Redbank"}
+}
+
+def toDate(dateStr: str):
+    items = dateStr.split("-")
+    return datetime.date(int(items[1]), int(items[2]), int(items[3]))
+
+class IpswichGovAuParser(HTMLParser):
+    def __init__(self):
+        super().__init__()
+        self._entries = []
+        self._state = None
+        self._level = 0
+        self._class = ""
+        self._li_level = 0
+        self._li_valid = False
+        self._span_level = 0
+        self._load_date = False
+        self._load_bin = False
+        self._loaded_date = None
+
+    @property
+    def entries(self):
+        return self._entries
+
+    def handle_endtag(self, tag):
+
+        if tag == "li":
+            self._li_level -= 1
+            self._loaded_date = None
+
+        if tag == "span":
+            self._span_level -= 1
+
+    def handle_starttag(self, tag, attrs):
+        
+        d = dict(attrs)
+        cls = d.get("class", "")
+        
+        if tag == "li":
+            self._li_level += 1
+            if self._li_level == 1 and cls == "WBD-result-item":
+                self._li_valid = True
+            else:
+                self._li_valid = False
+                self._loaded_date = None
+
+        if tag == "span":
+            self._span_level += 1
+            if self._li_valid and self._span_level == 1 and cls == "WBD-event-date":
+                self._load_date = True
+
+            if self._li_valid and self._span_level == 3 and cls == "WBD-bin-text":
+                self._load_bin = True
+
+
+    def handle_data(self, data):
+
+        if not self._li_valid:
+            return
+
+        if self._load_date:
+            self._load_date = False
+
+            items = data.strip().split("-")
+            self._loaded_date = datetime.date(int(items[0]), int(items[1]), int(items[2]))
+
+        if self._load_bin:
+            self._load_bin = False
+
+            if data == "Waste Bin":
+                self._entries.append(Collection(self._loaded_date, "Rubbish", icon="mdi:trash-can"))
```

Comment:
Please do not rename the original waste types. Aliases for waste types can be configured by user.
Use a dict to get the icon for known waste types.
Then we don't need the if statements any more :-)